contracts/reward/reward_test.go

```diff
@@ -149,53 +151,72 @@ func TestUpdateGovernanceParameterByEpoch(t *testing.T) {
 	// 2. update governance parameter with block number before epoch(30 in this test), it should not be updated
 	// 3. update governance parameter with block number after epoch(31 in this test), it should be updated
 	allocBlockRewardCache()
-	blockNumber := int64(1)
+	blockNumber := uint64(1)
 	epoch := uint64(30)
 	cnRatio := new(big.Int).SetUint64(34)
 	pocRatio := new(big.Int).SetUint64(54)
 	kirRatio := new(big.Int).SetUint64(12)
 	mintingAmount := uint64(9600000000)
 
 	//1. update governanceParameter with block number 1 and check if it is updated well
-	header := &types.Header{Number: big.NewInt(blockNumber)}
+	header := &types.Header{Number: big.NewInt(0).SetUint64(blockNumber)}
 	config := &params.ChainConfig{Istanbul: governance.GetDefaultIstanbulConfig(), Governance: governance.GetDefaultGovernanceConfig(params.UseIstanbul)}
 
 	config.Istanbul.Epoch = epoch
 	config.Governance.Reward.Ratio = "34/54/12"
 	config.Governance.Reward.MintingAmount = new(big.Int).SetUint64(mintingAmount)
-
-	governanceParameter := getRewardGovernanceParameters(config, header)
+	dbm := database.NewDBManager(&database.DBConfig{DBType: database.MemoryDB})
+	gov := governance.NewGovernance(config, dbm)
+	governanceParameter := getRewardGovernanceParameters(gov, header)
 
 	if governanceParameter.blockNum != 1 || governanceParameter.cnRewardRatio.Cmp(cnRatio) != 0 ||
 		governanceParameter.pocRatio.Cmp(pocRatio) != 0 || governanceParameter.kirRatio.Cmp(kirRatio) != 0 {
 		t.Errorf("GovernanceParameter is different")
 	}
 
+	blockNumber = 31
+	currentSet := governance.NewGovernanceSet()
+	delta := governance.NewGovernanceSet()
+	_, current, _ := gov.ReadGovernance(blockNumber)
+	currentSet.Import(current)
+	delta.SetValue(params.Ratio, "40/30/30")
+	gov.WriteGovernance(blockNumber-1, currentSet, delta)
```

Comment:
This also just question.
Why 'blockNumber-1' as a parameter to gov.WriteGovernance?

Reply:
I also wondering this. :) Could you write the reason? :)